osu.Game/Screens/Play/SongProgressGraph.cs

```diff
@@ -6,43 +6,207 @@
 using System.Diagnostics;
 using osu.Game.Rulesets.Objects.Types;
 using osu.Game.Rulesets.Objects;
+using osu.Framework.Timing;
+using osu.Game.Beatmaps.Timing;
 
 namespace osu.Game.Screens.Play
 {
     public class SongProgressGraph : SquareGraph
     {
+        private List<double> strains = new List<double>();
+
+        public List<double> Strains
+        {
+            get { return strains; }
+            set
+            {
+                for(int x = 0; x < value.Count; x++)
+                {
+                    if (x == 0)
+                    {
+                        strains.Add(value[x]);
+                        strains.Add(value[x]);
+                        Values.Add(value[x]);
+                        Values.Add(value[x]);
+                    }
+                    else
+                    {
+                        strains.Add(value[x]);
+                        Values.Add(value[x]);
+                    }
+                }
+            }
+        }
+
         private IEnumerable<HitObject> objects;
 
         public IEnumerable<HitObject> Objects
         {
+            get { return objects; }
             set
             {
                 objects = value;
 
-                const int granularity = 200;
-                Values = new int[granularity];
-
                 if (!objects.Any())
-                    return;
+                {
+                    for (int x = 0; x < Strains.Count; x++)
+                    {
```

Comment:
Remove brackets, can also use a `foreach` loop

Reply:
can't be `foreach` because i can't assign value to foreach variable

Reply:
Doesn't the following work?

```
foreach (var s in Strains)
    s = 0;
```

Reply:
yeah, compiler error goes if like there